src/main/java/org/wildfly/naming/client/remote/RemoteServerTransport.java

```diff
@@ -76,8 +79,13 @@ final class RemoteServerTransport implements RemoteTransport {
         this.messageTracker = messageTracker;
         this.localContext = localContext;
         this.configuration = new MarshallingConfiguration();
-        configuration.setVersion(version == 2 ? 4 : 2);
+        configuration.setVersion(version >= 2 ? 4 : 2);
         configuration.setClassResolver(classResolverFilter != null ? new FilterClassResolver(classResolverFilter) : new ContextClassResolver());
+        if (version < JAKARTAEE_PROTOCOL_VERSION && LATEST_VERSION >= JAKARTAEE_PROTOCOL_VERSION) {
+            // JNDI server uses JNDI PROTOCOL version 3 or above but JNDI client uses JNDI PROTOCOL version 2 or below
+            // so in this case we need to translate classes from JavaEE API to JakartaEE API and vice versa
+            configuration.setClassNameTransformer(ClassNameTransformer.JAVAEE_TO_JAKARTAEE);
+        }
```

Comment:
Should we add a logging here to alert the user of this transformer?  I think  this is an important step, and a log will help debug if anything is not going as expected.

Reply:
Very good idea @chengfang .